tools/toolchains/iar.py

```diff
@@ -126,17 +126,22 @@ def cc_extra(self, object):
         base, _ = splitext(object)
         return ["-l", base + '.s.txt']
 
-    def get_compile_options(self, defines, includes):
+    def get_compile_options(self, defines, includes, for_asm=False):
         opts = ['-D%s' % d for d in defines] + ['-f', self.get_inc_file(includes)]
-        config_header = self.get_config_header()
-        if config_header is not None:
-            opts = opts + ['--preinclude', config_header]
+        if for_asm:
+            # The assembler doesn't support '--preinclude', so we need to add
+            # the macros directly
+            opts = opts + ['-D%s' % d for d in self.get_config_macros()]
```

Comment:
Again, if you open Keil uVision, go to "Options for Target...", go to ASM tab and add a macro like MBED_TIMESTAMP=123232412.123232 or MBED_CONFIG_NSAPI_ETC={0xc0} then all compiles will fail and ARMCC will complain that invalid macro is specified. Knowing that this is an issue both for ARMCC and IAR we could filter these.